word_counter.rb

```diff
@@ -0,0 +1,27 @@
+class WordCounter
+  def initialize
+  end
+
+  def count(text)
+    parsed = text.downcase.gsub(/(?:[^\w\s\n'\-]|(?<=[^a-z])\-|\-(?=[^a-z]))+/, '').gsub(/[\n\s]+/, ' ').strip
+    words = parsed.split(' ')
+
+    list = {}
+
+    words.each do |word|
+      if list.key?(word)
+        list[word] += 1
+      else
+        list[word] = 1
+      end
```

Comment:
You can definitely get this done in a single line. Look into other ways of using .each.